src/property.ts

```diff
@@ -233,10 +233,7 @@ export class Property implements PropertyPath {
 				else
 					throw new Error(`Invalid property 'init' option of type '${getTypeName(options.init)}'.`);
 
-				const property = this;
-				this.initializer = function () {
-					return targetType.model.serializer.deserialize(this, initFn.call(this), property, new InitializationContext(true));
-				};
+				this.initializer = initFn;
```

Comment:
Store raw initializer function (result will be handled / transformed in `Property$init`)